src/main/scala/net/snowflake/spark/snowflake/io/SnowflakeRDD.scala

```diff
@@ -11,22 +11,39 @@ class SnowflakeRDD(
                     sc: SparkContext,
                     fileNames: List[String],
                     format: SupportedFormat,
-                    downloadFile: (String) => InputStream
+                    downloadFile: (String) => InputStream,
+                    expectedPartitionCount: Int
                   ) extends RDD[String](sc, Nil) {
 
-  @transient private val FILES_PER_PARTITION = 2
+  @transient private val MIN_FILES_PER_PARTITION = 2
 
   override def compute(split: Partition, context: TaskContext): Iterator[String] = {
     val stringIterator = new SFRecordReader(format)
+    stringIterator.setDownloadFunction(downloadFile)
```

Comment:
Seems like these functions can be part of the constructor.

Reply:
My initial idea is to integrate the download function into the constructor. But, it is some difficult, so I separate it from constructor. There are another two use cases that the InputStream is for some local files not from network. In this way, the code change is clearer, and lower risk. Currently, SFRecordReader can handle both Filename and InputStream.